### PR TITLE
Satoshi/advanced colour search

### PR DESCRIPTION
**Pull request by sgmbr**
Fixed issue #83 
Added react-bootstrap-typeahead to achieve good UX for multiple select and customised options rendering.

### CODE REVIEW

File: src/components/bandCombos/BandComboSearchForm.js [hunk 1]
```diff
@@ -11,7 +15,7 @@ class BandComboSearchForm extends Component {
       style: '',
       study_area: '',
       bird__status: '',
-      colours: '',
+      colours: [],
```
Comment (sgmbr) [inline src/components/bandCombos/BandComboSearchForm.js]: Store as an array of objects to easily manipulate select options. Convert to CSV when creating query.

File: src/components/bandCombos/BandComboSearchForm.js [hunk 2]
```diff
@@ -46,11 +47,63 @@ class BandComboSearchForm extends Component {
...
+    if (colour.value === 'white') {
+      style.border = '1px solid #ddd';
+    }
+    return (
+      <React.Fragment key={index}>
+        <span style={ style }>colour</span>
+        <span>{ colour.label }</span>
+      </React.Fragment>
+    );
+  }
```
Comment (sgmbr) [inline src/components/bandCombos/BandComboSearchForm.js]: A custom render for select options.

File: src/components/bandCombos/BandComboSearchForm.js [hunk 3]
```diff
@@ -46,11 +47,63 @@ class BandComboSearchForm extends Component {
...
+    return (
+      <Token
+        key={ index }
+        onRemove={ props.onRemove }
+        style={ style }
+      >
+        c
+      </Token>
+    );
   }
```
Comment (sgmbr) [inline src/components/bandCombos/BandComboSearchForm.js]: A custom render for selected

File: src/components/bandCombos/BandComboSearchForm.js [hunk 4]
```diff
@@ -46,11 +47,63 @@ class BandComboSearchForm extends Component {
...
+        label: colour.charAt(0).toUpperCase() + colour.slice(1),
+        value: colour,
+        hex: colours[colour],
+      }))
+      : [];
+
+    if (this.state.colours.length >= 4) {
+      colourOptions = this.state.colours;
+    }
+
```
Comment (sgmbr) [inline src/components/bandCombos/BandComboSearchForm.js]: List of options created based on colours.js. Limit selection to 4.

File: src/components/bandCombos/BandComboSearchForm.js [hunk 5]
```diff
@@ -46,11 +47,64 @@ class BandComboSearchForm extends Component {
...
+    if (colour.value === 'white') {
+      style.border = '1px solid #ddd';
+    }
+    return (
+      <Token
+        key={ index }
+        onRemove={ props.onRemove }
+        style={ style }
+      >
+        c
```
Comment (georgemoon) [inline src/components/bandCombos/BandComboSearchForm.js]: @sgmbr What's the relevance of the 'c'?

Comment (sgmbr) [reply src/components/bandCombos/BandComboSearchForm.js]: It was a placeholder for adding height to the rendered div. Changed from 'c' to 'col' to add some meaning.

**Pull request merged**